addon/mixins/in-viewport.js

```diff
@@ -88,7 +88,12 @@ export default Mixin.create({
   _setViewportEntered() {
     const scrollableArea = get(this, 'scrollableArea') ? document.querySelector(get(this, 'scrollableArea')) : null;
 
-    const element = get(this, 'element');
+    let element;
+    if (get(this, 'viewportElement')) {
+      element = document.querySelector(get(this, 'viewportElement'));
+    } else {
+      element = get(this, 'element');
+    }
```

Comment:
Looks like to `unobserver` the intersection observer, we would need to pass this element as well.  (in `_unbindListeners`)